### PR TITLE
ADD superstring type + unit tests

### PR DESCRIPTION
**Pull request by manuelCarlos**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
...
+
+- `Superstring` type as main attributed string constructor.
+
+### Changed
+
+- `NSAttributedString` is no longer extended with a  `init(@NSAttributedStringBuilder _ builder: () -> NSAttributedString)` initialiser.
+
+### Fixed
+
+- Build error when trying to create an attributed string using a single  `AString` invocation.
```
Comment (blessingLopes) [inline CHANGELOG.md]: Noice!

File: Sources/Superstring/Superstring.swift [hunk 2]
```diff
@@ -8,25 +8,11 @@
...
-
-extension NSAttributedString {
+    let attributedString: NSAttributedString
     
-    public convenience init(@NSAttributedStringBuilder _ builder: () -> NSAttributedString) {
-        self.init(attributedString: builder())
-    }
+    public init(@NSAttributedStringBuilder _ builder: () -> AttributedStringConvertible) {
+        attributedString = builder().attributedString
+     }
```
Comment (blessingLopes) [inline Sources/Superstring/Superstring.swift]: - [ ] align

File: Tests/SuperstringTests/AStringTests.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+class AStringTests: XCTestCase {
+    
+    func test_AString() {
+        let expected = NSAttributedString(string: "Superstring")
+        let result = AString("Superstring")
+        
+        XCTAssertTrue(result.attributedString.isEqual(expected))
+    }
+    
+    func test_AString_with_background_color() {
```
Comment (blessingLopes) [inline Tests/SuperstringTests/AStringTests.swift]: test_AString_with_red_background_color() would be better, imo

File: Tests/SuperstringTests/AStringTests.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        let expected = NSAttributedString(string: "Hello Superstring",
+                                          attributes: [.backgroundColor: Color.red])
+        let result = AString("Hello Superstring").backgroundColor(.red)
+        
+        XCTAssertTrue(result.attributedString.isEqual(expected))
+    }
+    
+    func test_String_with_multiple_attributes() {
```
Comment (blessingLopes) [inline Tests/SuperstringTests/AStringTests.swift]: More expressive name here to 😉

File: Tests/SuperstringTests/AStringTests.swift (hunk 3 above)
Comment (manuelCarlos) [reply Tests/SuperstringTests/AStringTests.swift]: Ok will do

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,18 @@
...
+
```
Comment (manuelCarlos) [inline CHANGELOG.md]: don't forget to add the links at the bottom

**Pull request merged**